### PR TITLE
watchlist first iteration

### PR DESCRIPTION
**Pull request by lucaimbrogno**
image

also figured out the prettier thing

### CODE REVIEW

File: packages/dashboard/src/components/lists/List.js [hunk 1]
```diff
@@ -19,100 +19,91 @@ import {
...
+        <td>{'$ ' + item.quote.amount}</td>
+        <td>
+          <div
+            style={{
+              width: '60%',
+              padding: '8px',
+              borderRadius: '6px',
+              background:
+                item.quote.previous_close < item.quote.amount
+                  ? 'lightgreen'
```
Comment (alexanderMontague) [inline packages/dashboard/src/components/lists/List.js:38]: using `lightgreen` and I presume the red rgba. I wouldnt mix rgba and variable defaults. The greens look a bit different from what we use in the smallStats, maybe use the same green + red there for consistency?

File: packages/dashboard/src/views/Watchlists.js [hunk 2]
```diff
@@ -21,60 +20,45 @@ import {
...
 
-  const checkWatchlist = () => {
-    console.log(watchlist)
+  const renderList = () => {
+    return watchlist.securities ? (
+      <List listTitle={'WST Watchlist'} items={watchlist.securities} />
+    ) : (
+      <h2 style={{ margin: 'auto' }}>Loading Watchlist Data...</h2>
+    )
   }
```
Comment (alexanderMontague) [inline packages/dashboard/src/views/Watchlists.js]: If you dont need to do any logic beforehand in the function, arrow functions let you simply return whatever they point to.
```suggestion
  const renderList = () =>
    watchlist.securities ? (
      <List listTitle={'WST Watchlist'} items={watchlist.securities} />
    ) : (
      <h2 style={{ margin: 'auto' }}>Loading Watchlist Data...</h2>
    )
```

File: packages/dashboard/src/views/Watchlists.js [hunk 3]
```diff
@@ -21,60 +20,45 @@ import {
...
     if (isLoggedIn) {
-      getWatchlist({tokens: JSON.stringify(user.tokens)})
+      getWatchlist({ tokens: JSON.stringify(user.tokens) })
     }
   }, [isLoggedIn])
 
-  const checkWatchlist = () => {
-    console.log(watchlist)
+  const renderList = () => {
+    return watchlist.securities ? (
```
Comment (alexanderMontague) [inline packages/dashboard/src/views/Watchlists.js]: Can you utilize the `watchlistLoading` here or whatever, and then specify more in the "loading" state? Ex. That flag will let you display the loading, and then when its not we could check if the response back succeeded and we return the List component, or theres an error and we can show an error state. I believe if theres an error watchlist will return our error object (check the saga tho to be sure)

**Review by alexanderMontague — CHANGES_REQUESTED** [review]
Few things, looks good though. I know this is a first iteration but we could probably consolidate the design a bit more to be in line with our current template.

File: packages/dashboard/src/components/lists/List.js (hunk 1 above)
Comment (lucaimbrogno) [reply packages/dashboard/src/components/lists/List.js:38]: yup good call. I was planning on playing with the colours, I'm also a bit colourly challenged when it comes to red/green.

Comment (alexanderMontague) [reply packages/dashboard/src/components/lists/List.js:38]: LOL #colourblind

File: packages/dashboard/src/views/Watchlists.js [hunk 4]
```diff
@@ -21,60 +20,56 @@ import {
...
-    console.log(watchlist)
+  const renderList = () => {
+    return isWatchlistLoading ? (
+      <h2 style={{ margin: 'auto' }}>Loading Watchlist Data...</h2>
+    ) : watchlist.securities ? (
+      <List listTitle={'WST Watchlist'} items={watchlist.securities} />
+    ) : watchlist.error ? (
+      <h2 style={{ margin: 'auto' }}>{watchlist.error.message}</h2>
+    ) : (
+      <></>
```
Comment (alexanderMontague) [inline packages/dashboard/src/views/Watchlists.js]: Can also return `null` and nothing will be rendered too.

File: packages/dashboard/src/components/lists/List.js [hunk 5]
```diff
@@ -19,100 +19,91 @@ import {
...
-  user,
-  historicQuotes,
-  globalSelectedAccount,
-}) => {
-  
-
-  
+const List = ({ items, listTitle }) => {
+  const renderListItems = () => {
+    return items.map(item => (
```
Comment (alexanderMontague) [inline packages/dashboard/src/components/lists/List.js:24]: Same here, map returns an array so forgoing the curly brackets and return statement is a little bit cleaner. Just for future use this is fine now.
```
const renderListItems = () => 
    items.map(item => ....
```

**Review by alexanderMontague — APPROVED** [review]

File: packages/dashboard/src/components/lists/List.js (hunk 5 above)
Comment (lucaimbrogno) [reply packages/dashboard/src/components/lists/List.js:24]: I'm not sure if im doing it wrong but it doesn't seem to work properly when I remove the return keyword in the examples you suggested

**Pull request merged**